source/common/stats/symbol_table_creator.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "common/stats/fake_symbol_table_impl.h"
+#include "common/stats/symbol_table_impl.h"
+
+namespace Envoy {
+namespace Stats {
+
+class SymbolTableCreator {
+public:
+  // Factory method to create SymbolTables. This is needed to help make it
+  // possible to flag-flip use of real symbol tables, and but ultimately
+  // should be removed.
+  static SymbolTablePtr makeSymbolTable(bool use_fake) {
+    use_fake_symbol_tables_ = use_fake;
+    return makeSymbolTable();
+  }
+
+  static SymbolTablePtr makeSymbolTable() {
+    if (use_fake_symbol_tables_) {
+      return std::make_unique<FakeSymbolTableImpl>();
+    }
+    return std::make_unique<SymbolTableImpl>();
+  }
+
+  // Sets whether fake or real symbol tables should be used. Tests that alter
```

Comment:
In prod code IMO this should only be set once and then we make sure it is never set again, and not set after it's been read. Can you sync up with @alyssawilk who has done similar stuff lately with runtime? I'm hoping maybe we can use a similar pattern here?

Reply:
I think you are specifically asking to switch from using a new Option to using source/common/runtime/runtime_features.h instead. Is that right?

I'm fine with that. I was just following the pattern for the new buffer impl in this PR. Happy to switch. @alyssawilk did we not use this for the new buffer impl because RuntimeFeatures wasn't ready yet at the time?

Reply:
Also... how does one enable a guarded feature in production? Is it possible to do so on the command-line?

Reply:
There are a few different things here:
1) Whether we use runtime or CLI, I just want to make sure this is set once and not set after it's been read to avoid bouncing/confusing values. Basically something like the asserts in `InjectableSingleton` (I think that's where @alyssawilk and I had this convo but I can't remember exactly right now).
2) In terms of using runtime vs. CLI, I think runtime is highly preferred, but I think in this case you have a circular dependency in that runtime itself has stats, so I don't think this will actually work and you have to use CLI if you want the entire process to be consistent. Right?
3) If we determine that we can use runtime, runtime can now be set via bootstrap and implicitly then via CLI (via config merge), in addition to FS, RTDS, etc. so there are meny options.